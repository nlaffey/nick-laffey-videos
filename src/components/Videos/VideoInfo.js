import React from 'react';

class VideoInfo extends React.Component {

    static toUpperCase(string) {
        if (string) {
            return VideoInfo.toUpperCase();
        } else {
            return null;
        }
    }

    render() {
        var data = this.props.videoData, list = <div>&nbsp;</div>;

        if (data) {
            list = <ul>
                <li><strong>{VideoInfo.toUpperCase(data.name)}</strong></li>
                <li>{data.date}</li>
                <li>{data.location}</li>
                <li>Video by {data.videoBy}</li>
                <li>Music by {data.musicBy}</li>
            </ul>
        }

        return <div className="info">
            {list}
        </div>

    }
}
export default VideoInfo;