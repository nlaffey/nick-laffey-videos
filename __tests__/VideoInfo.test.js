import renderer from 'react-test-renderer';
import React from 'react';
import VideoInfo from '../src/components/Videos/VideoInfo';
import LetItDieImage from '../src/img/letitdie.png';

it('renders correctly 1', ()=> {
    const videoData = {
        name: "Let It Die",
        image: LetItDieImage,
        location: "Shot in the Raring Building at the University of Minnesota",
        href: "https://www.youtube.com/watch?v=-0xqFsr1b0k",
        date: "April 2013",
        videoBy: "Meagan Nouis",
        musicBy: "Fiest"
    }

    const video = renderer.create(
        <VideoInfo videoData={videoData}/>
    ).toJSON();
    expect(video).toMatchSnapshot();
});

it('renders correctly 2', ()=> {
    const videoData = {
        name: null,
        image: LetItDieImage,
        location: 11111,
        href: 'asdf',
        date: 'foobar',
        videoBy: "long entrylong entrylong entrylong entrylong entrylong entrylong entrylong entrylong entry",
        musicBy: ""
    }

    const video = renderer.create(
        <VideoInfo videoData={videoData}/>
    ).toJSON();
    expect(video).toMatchSnapshot();
});