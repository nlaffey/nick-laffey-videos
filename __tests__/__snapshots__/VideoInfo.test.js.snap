exports[`test renders correctly 1 1`] = `
<div
  className="info">
  <ul>
    <li>
      <strong />
    </li>
    <li>
      April 2013
    </li>
    <li>
      Shot in the Raring Building at the University of Minnesota
    </li>
    <li>
      Video by 
      Meagan Nouis
    </li>
    <li>
      Music by 
      Fiest
    </li>
  </ul>
</div>
`;

exports[`test renders correctly 2 1`] = `
<div
  className="info">
  <ul>
    <li>
      <strong />
    </li>
    <li>
      foobar
    </li>
    <li>
      11111
    </li>
    <li>
      Video by 
      long entrylong entrylong entrylong entrylong entrylong entrylong entrylong entrylong entry
    </li>
    <li>
      Music by 
      
    </li>
  </ul>
</div>
`;
